Block/Checkout/Success.php

```diff
@@ -116,6 +141,50 @@ public function getAction()
         return json_encode($this->getOrder()->getPayment()->getAdditionalInformation('action'));
     }
 
+    public function showAdyenGiving()
+    {
+        return $this->adyenGivingEnabled() && $this->hasDonationToken();
+    }
+
+    public function adyenGivingEnabled(): bool
+    {
+        return (bool) $this->configHelper->adyenGivingEnabled($this->storeManager->getStore()->getId());
+    }
+
+    public function hasDonationToken()
+    {
+        return $this->getDonationToken() && 'null' !== $this->getDonationToken();
```

Comment:
I did it this way because the `->getAdditionalInformation('donationToken')` function could also return a string literal value of `'null'`, for some reason